src/cinder/linux/GstPlayer.cpp

```diff
@@ -177,8 +177,10 @@ gboolean checkBusMessages( GstBus* bus, GstMessage* message, gpointer userPlayer
 				gst_message_parse_state_changed( message, &old, &current, &pending );
 				
 				data.mCurrentState = current;
-				
 				g_print( "Pipeline state changed from : %s to %s\n", gst_element_state_get_name( old ), gst_element_state_get_name ( current ) );
+
+				if( current > GST_STATE_NULL && current <= GST_STATE_PLAYING ) data.mIsDone = false;

```

Comment:
Code style nit: can you place the contents of the conditional on a new line? Details [here](https://github.com/cinder/Cinder/blob/master/CONTRIBUTING.md#spacing-and-line-breaks).